### PR TITLE
SI-8081 unzip/unzip3 return wrong static type when applied to Arrays

### PR DESCRIPTION
**Pull request by Ichoran**
Added unzip and unzip3 methods to ArrayOps so that we get proper return
values.  (No tests due to unfavorable effort/reward ratio for an ongoing
test of something so simple.)

### CODE REVIEW

File: src/library/scala/collection/mutable/ArrayOps.scala [hunk 1]
```diff
@@ -107,6 +107,54 @@ trait ArrayOps[T] extends Any with ArrayLike[T, Array[T]] with CustomParalleliza
...
+    }
+    (a1, a2)
+  }
+  
+  /** Converts an array of triples into three arrays, one containing the elements from each position of the triple.
+   *  
+   *  @tparam T1    the type of the first of three elements in the triple
+   *  @tparam T2    the type of the second of three elements in the triple
+   *  @tparam T3    the type of the third of three elements in the triple
+   *  @param asPair an implicit conversion which asserts that the element type
```
Comment (xuwei-k) [inline src/library/scala/collection/mutable/ArrayOps.scala]: s/asPair/asTriple

Comment (gkossakowski) [inline src/library/scala/collection/mutable/ArrayOps.scala]: @Ichoran: could you fix that typo?

**Comment by gkossakowski** [comment]
Once the typo is fixed this will be good to go in.

**Comment by Ichoran** [comment]
Typo fixed.

**Comment by adriaanm** [comment]
LGTM by proxy then, merging

**Pull request merged**